apps/src/code-studio/components/progress/teacherPanel/SelectedStudentInfo.jsx

```diff
@@ -65,6 +66,31 @@ export default class SelectedStudentInfo extends React.Component {
     }
   };
 
+  renderPartners({partners, partnerCount}) {
+    // Three cases:
+    // - no known names: "3 other students"
+    // - exactly one partner whose name is known: "Student name"
+    // - all other cases: "Student name + 2" (with tooltip listing all known names)
+    if (partners.length === 0) {
```

Comment:
Super NIT: I'd prefer if `partners` was `partnerNames` just to clarify that we don't expect full partner objects in this array.